module/translation-deepl/src/Infrastructure/Provider/Decorator/TranslationDeeplProviderDecorator.php

```diff
@@ -0,0 +1,113 @@
+<?php
+
+/**
+ * Copyright © Bold Brand Commerce Sp. z o.o. All rights reserved.
+ * See license.txt for license details.
+ */
+
+declare(strict_types = 1);
+
+namespace Ergonode\TranslationDeepl\Infrastructure\Provider\Decorator;
+
+use Doctrine\DBAL\Connection;
+use Doctrine\DBAL\Query\QueryBuilder;
+use Ergonode\Core\Domain\ValueObject\Language;
+use Ergonode\TranslationDeepl\Infrastructure\Provider\TranslationDeeplProvider;
+use Ergonode\TranslationDeepl\Infrastructure\Provider\TranslationDeeplProviderInterface;
+use Ramsey\Uuid\Uuid;
+
+/**
+ */
+class TranslationDeeplProviderDecorator implements TranslationDeeplProviderInterface
```

Comment:
Cache handling should be move to separate class base on CacheInterface.
CacheInterface method get(string $key), set(string $key, $data), has(string $key): bool